server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanDynamicPropertyCache.java

```diff
@@ -0,0 +1,118 @@
+import org.hawkular.apm.api.utils.PropertyUtil;
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.DynamicPropertyService;
+import org.hawkular.apm.server.api.services.ObservableProperty;
+import org.infinispan.Cache;
+import org.infinispan.manager.CacheContainer;
+
+/**
+ * @author Pavol Loffay
+ */
+public class InfinispanDynamicPropertyCache implements DynamicPropertyService, ServiceLifecycle {
+
+    private static final Logger log = Logger.getLogger(InfinispanSpanCache.class.getName());
+
+
+    protected static final String CACHE_NAME = "property";
+
+    @Resource(lookup = "java:jboss/infinispan/APM")
+    private CacheContainer cacheContainer;
+
+    private Cache<String, ObservableProperty> propertyCache;
+
+
+    @Override
+    @PostConstruct
+    public void init() {
+        // If cache container not already provisions, then must be running outside of a JEE
+        // environment, so create a default cache container
+        if (cacheContainer == null) {
+            log.fine("Using default cache");
+            setPropertyCache(InfinispanCacheManager.getDefaultCache(CACHE_NAME));
+        } else {
+            log.fine("Using container provided cache");
+            setPropertyCache(cacheContainer.getCache(CACHE_NAME));
+        }
+    }
+
+    @Override
+    public ObservableProperty<?> get(String tenantId, String id) {
+        ObservableProperty<?> observableProperty = propertyCache.get(id);
+
+        if (observableProperty == null) {
+            observableProperty = getFromSystemOrEnvVariables(id);
+            if (observableProperty == null) {
+                return null;
+            }
+
+            try {
+                store(tenantId, Collections.singletonList(observableProperty));
+            } catch (CacheException ex) {
+                log.severe(String.format("Could not store system or env property: %s, to cache: ", id));
+            }
+        }
+
+        log.finest("Get communication details [id=" + id + "] = " + observableProperty);
+        return observableProperty;
+    }
+
+    @Override
+    public void store(String tenantId, List<ObservableProperty> observableProperties) throws CacheException {
+        if (cacheContainer != null) {
+            propertyCache.startBatch();
+        }
+
+        for (ObservableProperty observableProperty : observableProperties) {
+            log.finest("Store communication details [id=" + observableProperty.getId() + "]: " + observableProperty);
+            propertyCache.put(observableProperty.getId(), observableProperty);
+        }
+
+        if (cacheContainer != null) {
+            propertyCache.endBatch(true);
+        }
+    }
+
+    public Cache<String, ObservableProperty> getPropertyCache() {
+        return propertyCache;
+    }
+
+    public void setPropertyCache(Cache<String, ObservableProperty> propertyCache) {
+        this.propertyCache = propertyCache;
```

Comment:
I don't understand why you'd want to provide a setter for the property cache, given that you construct one at `@PostConstruct`. If you want to let the client code to specify the `propertyCache`, then I'd provide a constructor that accepts it, instead of letting the caller code to change it at will. 

If you really need a setter for it, then you'd probably sanity check it before accepting. For instance, checking if it's null.